hack/make-release.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# set vars 
+if [ -n "$1" ]; then
+    DOCKER_REPO=$1
+else
+    echo "Must provide a docker repo"
+    exit 
+fi
+
+if [ -n "$2" ]; then
+    TAG=$2
+else
+    echo "Must provide a version tag."
+    exit 
+fi
+
+if [ -n "$3" ]; then
+    BRANCH=$3
+else
+    echo "Must provide a git branch."
+    exit 
+fi
+
+# iterate over all services: build + push image, inject tag into k8s manifest. 
+for dir in ./src/*/    
+do
+    # docker build + push 
+    svcname=$(basename $dir)
+    image="$DOCKER_REPO/$svcname:$TAG"
+    echo "Building and pushing $image..." 
+    docker build -t $image -f $dir/Dockerfile $dir 
+    docker push $image 
+
+    # inject new tag into the relevant k8s manifest
+    pattern=".*image:.*"
+    replace="        image: $image"
```

Comment:
this number of spaces look hacky and fragile. I think you need to use groups to replace only matches with parantheses.

Reply:
Not 100% sure I know what you mean here. Do you mean capture groups? Why parentheses?